substrate/frame/staking/src/mock.rs

```diff
@@ -881,7 +881,12 @@ pub(crate) fn balances(who: &AccountId) -> (Balance, Balance) {
 pub(crate) fn stake_tracker_sanity_tests() -> Result<(), &'static str> {
 	use sp_staking::StakingInterface;
 
-	assert_eq!(Nominators::<Test>::count() + Validators::<Test>::count(), VoterBagsList::count());
+	assert_eq!(
+		Nominators::<Test>::count() + Validators::<Test>::count(),
+		VoterBagsList::iter()
+			.filter(|v| Staking::status(&v) != Ok(StakerStatus::Idle))
```

Comment:
may be better to reduce it to a tuple of (nominator_count, validator_count) and then assert on each count separately.

Reply:
To expand, you can have assertions for count of nominators and validators separately.

Reply:
Never mind what I said, its wrong.